queen-of-versions/src/main/java/com/infinum/queenofversions/GoogleInAppUpdateSuccessListener.java

```diff
@@ -0,0 +1,113 @@
+        UpdaterStateCallback flexibleListener, String versionCode, InstallStateUpdatedListener installStateUpdatedListener,
+        UpdaterCallback updaterCallback,
+        GoogleInAppUpdateFlexibleHandler handler) {
+        this.requestCode = requestCode;
+        this.activity = activity;
+        this.isMandatory = isMandatory;
+        this.appUpdateManager = appUpdateManager;
+        this.flexibleStateListener = flexibleListener;
+        this.installStateUpdatedListener = installStateUpdatedListener;
+        this.updaterCallback = updaterCallback;
+        this.handler = handler;
+        this.appVersionCode = versionCode;
+    }
+
+
+    //TODO we have to agree on what to do in specific version cases
+    @Override
+    public void onSuccess(AppUpdateInfo appUpdateInfo) {
+        if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE) {
+            if(checkVersionCode(appUpdateInfo)){
+                startUpdate(appUpdateInfo,isMandatory);
+            }else {
+                startUpdate(appUpdateInfo,false);
+            }
+        } else {
+            noUpdate();
+        }
+    }
+
+    private void startUpdate(AppUpdateInfo appUpdateInfo,boolean isMandatory) {
+        if (isMandatory) {
+            startImmediateFlow(appUpdateInfo);
+        } else {
+            startFlexibleFlow(appUpdateInfo);
+        }
+        appUpdateManager.registerListener(installStateUpdatedListener);
+    }
+
+    private void noUpdate() {
+        flexibleStateListener.onNoUpdate();
+    }
+
+    private void startFlexibleFlow(AppUpdateInfo appUpdateInfo) {
+        try {
+            appUpdateManager.startUpdateFlowForResult(
+                appUpdateInfo,
+                AppUpdateType.FLEXIBLE,
+                activity,
+                requestCode
+            );
+        } catch (IntentSender.SendIntentException e) {
+            updaterCallback.onError(e);
+        }
+    }
+
+    private void startImmediateFlow(AppUpdateInfo appUpdateInfo) {
+        try {
+            appUpdateManager.startUpdateFlowForResult(
+                appUpdateInfo,
+                AppUpdateType.IMMEDIATE,
+                activity,
+                requestCode
+            );
+            registerImmediateResumeFlow();
+        } catch (IntentSender.SendIntentException e) {
+            updaterCallback.onError(e);
+        }
+    }
+
+    private void registerImmediateResumeFlow() {
+        activity.getApplication().registerActivityLifecycleCallbacks(
+            new AppCallbacks(requestCode, activity, appUpdateManager, updaterCallback, flexibleStateListener, handler)
+        );
+    }
+
+    private boolean checkVersionCode(AppUpdateInfo appUpdateInfo) {
+        Integer versionCode;
+        if (appVersionCode == null) {
+            return false;
+        } else {
+            versionCode = Integer.parseInt(appVersionCode);
```

Comment:
This can throw if appVersionCode is not a valid integer. As this is from the config should handle that case and probably report an error.